cfy_manager/components/rabbitmq/rabbitmq.py

```diff
@@ -273,38 +278,29 @@ def _possibly_add_hosts_entries(self):
             logger.info('Updated /etc/hosts')
 
     def _generate_rabbitmq_certs(self):
-        broker_cert_path = config[RABBITMQ]['broker_cert_path']
-        broker_key_path = config[RABBITMQ]['broker_key_path']
+        supplied = self.use_supplied_certificates(
+            cert_destination=BROKER_CERT_LOCATION,
+            key_destination=BROKER_KEY_LOCATION,
+            ca_destination=BROKER_CA_LOCATION,
+            owner='rabbitmq',
+            group='rabbitmq',
+        )
 
-        if broker_cert_path and broker_key_path:
+        if supplied:
             logger.info('Using supplied certificates.')
-            if not config[RABBITMQ]['broker_ca_path']:
-                logger.info(
-                    'Broker CA path not supplied, assuming self-signed.'
-                )
-                config[RABBITMQ]['broker_ca_path'] = broker_cert_path
             return
-        elif broker_cert_path or broker_key_path:
-            raise ValidationError(
-                'Both broker_cert_path and broker_cert_key must be '
-                'provided if one of these settings is provided.'
-            )
         else:
-            config[RABBITMQ]['broker_cert_path'] = (
-                '/etc/cloudify/ssl/rabbitmq_cert.pem'
-            )
-            config[RABBITMQ]['broker_key_path'] = (
-                '/etc/cloudify/ssl/rabbitmq_key.pem'
-            )
+            config[RABBITMQ]['cert_path'] = BROKER_CERT_LOCATION
+            config[RABBITMQ]['key_path'] = BROKER_KEY_LOCATION
 
         logger.info('Generating rabbitmq certificate...')
 
         if self._installing_manager():
             has_ca_key = certificates.handle_ca_cert()
-            config[RABBITMQ]['broker_ca_path'] = constants.CA_CERT_PATH
+            config[RABBITMQ]['ca_path'] = constants.CA_CERT_PATH
         else:
             has_ca_key = False
-            config[RABBITMQ]['broker_ca_path'] = broker_cert_path
+            config[RABBITMQ]['ca_path'] = config[RABBITMQ]['cert_path']
```

Comment:
Shouldn't that be `BROKER_CA_LOCATION` in case a manager is not being installed together with it?
Like this it just takes the `ca_path` and `cert_path` to be the same certificate

Reply:
That's because it only does this if (it's not installing a manager) AND (certificates weren't supplied).

In this case, we'll generate a self-signed certificate, meaning that anything that wants to grab the cert that is needed to trust rabbit has to grab the cert rather than a separate ca one.

Reply:
Got it, maybe add a comment there just for future reference?

Reply:
Comment added.